integration-tests/bats/remotes.bats

```diff
@@ -2075,7 +2075,8 @@ SQL
     [ "$status" -eq 0 ]
     [[ "$output" =~ "Everything up-to-date." ]] || false
 
-    run dolt branch --track direct feature origin/other
+    # TODO: this command fails on GitHub requiring `--track=direct`, when both branch name and starting point name are defined, but Dolt allows both formats.
+    run dolt branch feature --track direct origin/other
```

Comment:
It would be good to make sure we have test coverage over all the forms you listed in the PR description. 

I saw these two: 
- git branch --track request origin/feature
- git branch feature --track=direct origin/feature

But I don't think I saw these covered: 
- git branch feature --track origin/feature
- git branch --track=direct feature origin/feature